src/main/java/wolox/training/models/Book.java

```diff
@@ -78,14 +79,16 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
-        genre = genre;
+        Preconditions.checkNotNull(genre);
+        this.genre = genre;
     }
 
     public String getAuthor() {
         return author;
     }
 
     public void setAuthor(String author) {
+        Preconditions.checkNotNull(author);
```

Comment:
The same as the previous comment.

Reply:
I put the messages and added it like constants.